src/lib/functions/categoryFrom.ts

```diff
@@ -1,12 +1,19 @@
 import { CategoryEnum } from '$lib/types/category';
 
 export function categoryFrom(query: string): string | undefined {
-	for (const category in CategoryEnum) {
-		if (query.startsWith(`!${category}`)) {
-			return category;
+	let gottenCategory: string = 'general';
```

Comment:
Maybe name it "categoryFromQuery" or "cat" (the latter is used in backend when reading from query)